plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/editor/autoedit/JSDocEditStrategy.java

```diff
@@ -0,0 +1,114 @@
+import org.eclipse.n4js.n4JS.N4MethodDeclaration;
+import org.eclipse.n4js.n4JS.Script;
+import org.eclipse.n4js.ui.editor.N4JSDocument;
+import org.eclipse.n4js.ui.organize.imports.XtextResourceUtils;
+import org.eclipse.xtext.nodemodel.ILeafNode;
+import org.eclipse.xtext.nodemodel.util.NodeModelUtils;
+import org.eclipse.xtext.ui.editor.autoedit.CommandInfo;
+import org.eclipse.xtext.ui.editor.autoedit.MultiLineTerminalsEditStrategy;
+
+public class JSDocEditStrategy extends MultiLineTerminalsEditStrategy {
+	private final String indentationString;
+	private final static String PARAMSTR = "@param ";
+	private final static String RETURNSTR = "@return ";
+
+	public JSDocEditStrategy(String leftTerminal, String indentationString, String rightTerminal) {
+		super(leftTerminal, indentationString, rightTerminal, false);
+		this.indentationString = indentationString;
+	}
+
+	/**
+	 * Expects the cursor to be in the same line as the start terminal puts any text between start terminal and cursor
+	 * into a separate newline before the cursor. puts any text between cursor and end terminal into a separate newline
+	 * after the cursor. puts the closing terminal into a separate line at the end. adds a closing terminal if not
+	 * existent.
+	 */
+	@Override
+	protected CommandInfo handleCursorInFirstLine(IDocument document, DocumentCommand command, IRegion startTerminal,
+			IRegion stopTerminal) throws BadLocationException {
+		CommandInfo newC = new CommandInfo();
+		List<String> retTypeAndfparNames = getRetTypeAndparNamesAsList(document, startTerminal);
+		String paramString = "";
+		String returnType = "";
+		if (retTypeAndfparNames.size() > 0) {
+			returnType = indentationString + RETURNSTR + "{" + retTypeAndfparNames.get(0) + "}";
+		}
+		if (retTypeAndfparNames.size() > 1) {
+			for (int i = 1; i < retTypeAndfparNames.size(); i += 2) {
+				paramString += command.text + indentationString + PARAMSTR + "{" + retTypeAndfparNames.get(i) + "} "
+						+ retTypeAndfparNames.get(i + 1);
+			}
+		}
+		newC.isChange = true;
+		newC.offset = command.offset;
+		newC.text += command.text + indentationString;
+		newC.cursorOffset = command.offset + newC.text.length();
+		if (stopTerminal == null && atEndOfLineInput(document, command.offset)) {
+			newC.text += command.text + getRightTerminal();
+		}
+		if (stopTerminal != null && stopTerminal.getOffset() >= command.offset
+				&& util.isSameLine(document, stopTerminal.getOffset(), command.offset)) {
+			String string = document.get(command.offset, stopTerminal.getOffset() - command.offset);
+			if (string.trim().length() > 0)
+				newC.text += string.trim();
+			if (!(retTypeAndfparNames.size() == 0)) {
+				newC.text += paramString + command.text + returnType + command.text;
+			} else {
+				newC.text += command.text;
+			}
+
+			newC.length += string.length();
+		}
+		return newC;
+	}
+
+	private List<String> getRetTypeAndparNamesAsList(IDocument document, IRegion startTerminal) {
```

Comment:
Add javadoc... where is the return type stored here?
Small typo (or rename to): getReturnTypeAndParameterNames (asList is clear because of the return type of this method)